### DISCUSSION TITLE
Builder image user and root permission

### DISCUSSION THREAD

**Discussion by RATANAJANGIR** [buildpacks/community#219] [General]
Hi all,

Inside the builder image container ( generated using pack builder builder.toml  command), I am running cnb/lifecycle/detector command but as a **root** user (it's not giving permission to run as root) and with **cnb**  getting error  **ERROR: failed to write buildpack group: open group.toml: permission denied**.

-> cnb/lifecycle/builder
ERROR: failed to read buildpack group: open group.toml: no such file or directory

```
[stack]
  build-image = "docker.io/paketobuildpacks/build:1.2.63-base-cnb"
  id = "io.buildpacks.stacks.bionic"
  run-image = "index.docker.io/paketobuildpacks/run:base-cnb"
  run-image-mirrors = ["gcr.io/paketo-buildpacks/run:base-cnb"]
```
My goal to just detect source code and build artifact that why I am running directly lifecycle. Please help me to resolve this issue. @natalieparellano

**Comment by natalieparellano** [comment]
@RATANAJANGIR by default the lifecycle will try to write group.toml to the layers directory so my guess is that the permissions on `/layers` are too restrictive for the cnb user. Are you able to check?

**Comment by RATANAJANGIR** [comment]
@natalieparellano Yes tried on cnb user but getting permission denied error.  is there way to run single lifecycle "cnb/lifecycle/detector" ??. Please suggest me to run single lifecycle, I also tried with creating different user but it also didn't worked.

**Reply by natalieparellano** [reply]
Could you try making the `/layers` dir owned by the cnb user? The "analyzer" does this here: https://github.com/buildpacks/lifecycle/blob/82da83291abf997478200bd9d3135b5ac8245349/cmd/lifecycle/analyzer.go#L94

**Reply by RATANAJANGIR** [reply]
@natalieparellano Thankyou, I am able to run. Can you tell me why I am getting connect timed issue, it failed to downloads some of maven dependencies , I have added my company certificates in layers directory,
can you please tell me where is looking the certificates and the stack build environment is creating env inside the container env? So its like one build env inside some env, is that mean I should add certificates inside  stack build or run image also?? but I tried both way. 

Here i am running /cnb/lifecycle/builder command.

![image](https://github.com/buildpacks/community/assets/50768957/a312eb63-fa9c-4f04-8ca4-79b2316c6dfc)

**Reply by natalieparellano** [reply]
@RATANAJANGIR you might try including the Paketo [ca-certificates buildpack](https://github.com/paketo-buildpacks/ca-certificates) in your build, it makes use of `SSL_CERT_DIR` and `SSL_CERT_FILE` environment variables to load the certs.
